alfalfa_worker/jobs/step_run_base.py

```diff
@@ -1,15 +1,82 @@
-import datetime
 import os
+from dataclasses import dataclass
+from datetime import datetime, timedelta
+from enum import auto
 
 from influxdb import InfluxDBClient
 
-from alfalfa_worker.lib.enums import RunStatus
-from alfalfa_worker.lib.job import (
-    Job,
+from alfalfa_worker.lib.constants import DATETIME_FORMAT
+from alfalfa_worker.lib.enums import AutoName, RunStatus
+from alfalfa_worker.lib.job import Job, message
+from alfalfa_worker.lib.job_exception import (
     JobException,
-    JobExceptionSimulation,
-    message
+    JobExceptionSimulation
 )
+from alfalfa_worker.lib.utils import to_bool
+
+
+class ClockSource(AutoName):
+    INTERNAL = auto()
+    EXTERNAL = auto()
+
+
+@dataclass
+class Options:
```

Comment:
Consolidated options for `StepRunBase` extending jobs into one place.

Reply:
TODO: document each of these options better

Reply:
Maybe that is an opportunity to consolidate the class hierarchy a little?